New PCB and big firmware update

sembruk opened the pull request:
Fix #61, fix #63, fix #68, fix #69, fix #75, fix #81, fix #84.

mars2860 committed:
Схема и плата версия 3:
1. Исправил подключение питания для DS3231
2. UART и SPI для программирования вынес на один разъём
3. Добавил цепь для измерения напряжения на батарее
4. Добавил тестовых точек
5. Добавил конденсатор на цепь DTR
6. Добавил фильтр на цепь RST

mars2860 committed:
Убрал с печатной платы области металлизации, иначе они экранируют антенну

sembruk committed:
Merge branch 'master' into develop

Manual fixed conflicts:
	Base station/firmware/BaseStation/BaseStation.ino
	Base station/hardware/prod/old/PCB.JPG
	Base station/hardware/prod/old/Scheme.jpg
	Base station/hardware/prod/sportiduino-base-assembly.jpg
	Base station/hardware/prod/sportiduino-base-gerber.zip
	Base station/hardware/prod/sportiduino-base-scheme.jpg
	Base station/hardware/prod/v1/sportiduino-base-v1-pcb.jpg
	Base station/hardware/prod/v1/sportiduino-base-v1-scheme.jpg
	Base station/hardware/prod/v2/sportiduino-base-v2-gerber.zip
	Base station/hardware/prod/v2/sportiduino-base-v2-pcb.jpg
	Base station/hardware/prod/v2/sportiduino-base-v2-scheme.jpg
	hardware/BaseStation/prod/old/PCB.JPG
	hardware/BaseStation/prod/old/Scheme.jpg
	hardware/BaseStation/prod/old/Station.zip
	hardware/BaseStation/prod/sportiduino-base-assembly.jpg
	hardware/BaseStation/prod/sportiduino-base-gerber.zip
	hardware/BaseStation/prod/sportiduino-base-scheme.jpg

sembruk committed:
Base station scheme: added 24LC256 memory

sembruk committed:
Updated v3m PCB

sembruk committed:
Updated PCB

sembruk committed:
Fix typo

sembruk committed:
PCB v3m: added 3d view and fixed edge line width

sembruk committed:
Base station PCB: updated assembly footprints positions

sembruk committed:
Fix #73

sembruk committed:
Change version to 2.6.3

sembruk committed:
Merge branch 'fix-issue-73' into v3

sembruk committed:
Updated pins for BS v3
and checkBattery refactoring

sembruk committed:
Reed switch interrupt for wake up

sembruk committed:
Reed switch: debug with hw

sembruk committed:
Fix power consumption bag in sleepmode for v3 pcb

sembruk committed:
Added ADC_IN and ADC_ENABLE pins

sembruk committed:
Added CardType enum

sembruk committed:
Refactoring

sembruk committed:
3-bytes version of MS

sembruk committed:
Master station refactoring

sembruk committed:
3-bytes version of BS

sembruk committed:
External EEPROM (I2C) handling (WiP)
and some refactoring

sembruk committed:
Write timestamps to dump master card

sembruk committed:
Master station: debug after refactoring

sembruk committed:
BS: debug timestamps log (external eeprom)

sembruk committed:
New log-card format (with timestamps)

sembruk committed:
Debug log-card writing/reading

mars2860 commented on firmware/BaseStation/BaseStation.ino:
```diff
@@ -9,65 +9,70 @@
 
 //-------------------------------------------------------------------
 // HARDWARE
+    
```
Не пойму, зачем вынес из #if HW_VERS? Потому что одинаково для всех версий плат?

sembruk replied:
Да, дублирование смущает.

mars2860 commented on firmware/BaseStation/BaseStation.ino:
```diff
@@ -325,8 +320,19 @@ void loop() {
 #endif
 
+#ifdef USE_REED_SWITCH
+    if(mode == MODE_SLEEP) {
```
Как идея, можно добавить, например, MODE_STORAGE в котором не будет работать RFID, а станция будет будиться часами или герконом. А MODE_SLEEP оставить как есть

mars2860 commented on firmware/BaseStation/BaseStation.ino:
```diff
@@ -445,7 +453,9 @@ void serialFuncReadInfo(byte *data, byte dataSize) {
     sendData[pos++] = SERIAL_RESP_INFO;
     // Write version
-    sendData[pos++] = VERS;
+    sendData[pos++] = HW_VERS;
```
Вроде 1 байта ещё хватало, а это придётся теперь SportiduinoPQ править

sembruk replied:
Максимальная версия major прошивки — 15, minor — 3, что уже для мало.
SportiduinoPQ из ветки `dev` уже умеет распознавать однобайтовые версии старого формата (1.0 — 1.4), текущего (2.6) и нового трёхбайтового.

mars2860 commented on firmware/BaseStation/BaseStation.ino:
```diff
@@ -709,11 +728,108 @@ uint16_t getMarkLogEnd() {
 }
 
+void i2cEepromWritePunch(uint16_t cardNum) {
+    pinMode(I2C_EEPROM_VCC, OUTPUT);
```
Это можно в setup перенести

mars2860 commented on firmware/BaseStation/BaseStation.ino:
```diff
@@ -748,115 +864,112 @@ bool checkBattery(bool beepEnabled) {
+            break;
+        case MASTER_CARD_READ_DUMP:
+#ifdef USE_I2C_EEPROM
+            processDumpMasterCardWithTimestamps(masterCardData, sizeof(masterCardData));
```
Может быть будет лучше писать лог одновременно и в EEPROM атмеги и во внешнюю EEPROM (при наличии). Т.е. во внутренний EEPROM лог отметок пишется всегда, даже когда есть внешняя EEPROM. Для снятия лога из внешней EEPROM сделать свой мастер-чип. А то так в SportiduinoPQ надо будет придумывать какую-нибудь галку есть у тебя EEPROM или нет. А если будет отдельный мастер-чип для внешней EEPROM, то не надо будет текущий функционал по логу отметок переделывать в SportiduinoPQ и в прошивке. Всего лишь добавить новый мастер-чип, который не будет влиять на существующий

sembruk replied:
SportiduinoPQ из ветки `dev` автоматически распознаёт лог с отметками времени и без. Галочка не нужна, всё просто.

mforis commented:
Ребят, сделайте, чтобы при выходе из сна база подавала сигнал, а потом производила отметку. Спасибо!

sembruk committed:
Voltage measurement (#63)

sembruk committed:
Old checkBattery for HW v1 and v2

sembruk committed:
Fix problem with ext EEPROM reading
and improvements

mforis commented:
Вот я ещё что подумал, сделайте возможность использование геркона в первой и второй версии плат.

sembruk committed:
MS: fix dump card reading
There were bad timestamps in output

sembruk committed:
We can use reed switch only if appropriate pin is defined

sembruk commented:
> Вот я ещё что подумал, сделайте возможность использование геркона в первой и второй версии плат.

Для использования геркона, схемы измерения напряжения питания и даже внешней EEPROM в платах v1 и v2 достаточно определить соответствующие пины в начале файла исходого кода прошивки.
А для версии v3 наоборот можно определения этих пинов убрать. Тогда дополнительные компонетны (геркон, память) не нужно паять на плату. Станция будет работать как прежде.

sembruk committed:
New settings format: Configuration structure

sembruk committed:
Read/write config

sembruk committed:
Rfid class and MS fw refactoring

sembruk committed:
Added makefile (Arduino-Makefile) for MS

sembruk committed:
BS: makefile and using new Rfid class

sembruk committed:
Debug rfid.cardWrite()

sembruk committed:
PCB: merge v3m into v3, remove RC filter from RST line

sembruk committed:
PCB: increased pads length for DS3231

sembruk committed:
Disable RFID module to prevent bad impact on voltage measurements
Added 'port' option in makefiles

sembruk committed:
Updated BS PCB

sembruk committed:
PCB: move battery voltage measurement circuit and remove 3V3-GND loops

sembruk committed:
Updated hardware images (scheme, pcb) and links

sembruk committed:
Clear config from EEPROM at first start

sembruk committed:
Added SerialProtocol class

sembruk committed:
Debug SerialProtocol class

sembruk committed:
BS set time: fourth beep for sync

sembruk committed:
BS: serial protocol like MS serial protocol

sembruk committed:
Don't set UART to output-low in sleep

sembruk committed:
Check battery voltage at startup

sembruk committed:
Erase I2C EEPROM at first startup

sembruk committed:
Prevent continuous waking up by UART RX pin undefined state

sembruk committed:
Check battery at waking up (#63, #81)

sembruk committed:
Don't overwrite last mark in full card (#84)

sembruk committed:
Deinitialize time and station number master cards by BS

sembruk committed:
MS: set RFID antenna gain by serial and store one in EEPROM

sembruk committed:
Debug with hw

sembruk committed:
Debug mode and buzzer frequency as additional make options

sembruk committed:
Min battery voltage now is 3.4 V

sembruk committed:
MS: read settings cmd

sembruk committed:
More build options for make command

sembruk committed:
Fix mistakes in PCB gerber files

sembruk committed:
Build for PCB v1 and v2

sembruk committed:
Merge branch 'master' into v3

sembruk committed:
Fix punches backup reading

sembruk committed:
Change version to 3.7.0-beta.1

sembruk committed:
Led blinking at writing backup card

sembruk committed:
Fix led blinking at writing backup card

sembruk committed:
Updated BS stickers for v3

sembruk committed:
Change readme files encoding to UTF-8

sembruk committed:
Change minimum battery voltage to 3.6 V

sembruk committed:
POLL_CARDS_IN_SLEEP_MODE option
Power on RC522 every 25 secs as in old firmware if the option enabled (default)

sembruk committed:
Fix wake up process
"make nopoll" defines NO_POLL_CARDS_IN_SLEEP_MODE option.
Tested with PCB v3

sembruk committed:
Changlelog for v3.7.0

sembruk committed:
Merge branch 'master' into v3

sembruk committed:
Update Readme (Ru and En)

sembruk committed:
Update Readme (Ru and En)

sembruk committed:
MS: fix participant card reading (fix #89)

sembruk committed:
Clear

sembruk committed:
BS PCB v3 2020-03-23: removed dual diode VD2 from measurement circuit

sembruk committed:
Update Ru User Manual

sembruk committed:
Ru docs update

sembruk committed:
Ru docs update

sembruk committed:
Ru docs update

sembruk committed:
Ru docs update

sembruk committed:
Ru docs update

sembruk committed:
Ru docs update

sembruk committed:
Don't check password at serial BS communication
